src/lazy/text/parse_result.rs

```diff
@@ -155,6 +146,19 @@ impl<'data> From<InvalidInputError<'data>> for IonError {
             message.push_str(" while ");
             message.push_str(label.as_ref());
         }
+        message.push_str("; buffer: ");
+        let input = invalid_input_error.input;
+        let buffer_text = if let Ok(text) = invalid_input_error.input.as_text() {
+            // TODO: This really should be graphemes instead of chars()
```

Comment:
If we're being pedantic... it should be _grapheme clusters_, right? 😉 

That being said, I don't think this really needs to be a "todo" item. It's unclear how valuable this is given that this message is already not going to contain the full text because of the truncation. From what I can tell, the ability to manipulate graphemes was removed from the standard library because of the size of the unicode tables and because it was not good for the std library to be coupled to a specific Unicode version.

Reply:
> If we're being pedantic... it should be grapheme clusters, right? 😉

Yep! 😄 

> That being said, I don't think this really needs to be a "todo" item [...]

I was concerned the code might do something weird if the `32` byte limit fell in a boundary between unicode scalars. However, on closer inspection of the `char` documentation, I see that:

> USVs are also the exact set of values that may be encoded in UTF-8. Because char values are USVs and str values are valid UTF-8, it is safe to store any char in a str or read any character from a str as a char.

so it should always produce a legal (if misleading) string if it's truncated.